admin-console/src/main/webapp/src/components/SystemBehaviorGroups/CreateEditBehaviorGroupModal.tsx

```diff
@@ -0,0 +1,85 @@
+    FormSelect,
+    FormSelectOption,
+    HelperText, HelperTextItem, Modal, ModalVariant, TextInput } from '@patternfly/react-core';
+import React from 'react';
+
+import { BehaviorGroup } from '../../types/Notifications';
+
+interface CreateEditModalProps {
+    isEdit: boolean;
+    showModal: boolean;
+    isOpen: boolean;
+    initialSystemBehaviorGroup?: Partial<BehaviorGroup>;
+    isLoading: boolean;
+    onClose: () => void;
+    onSubmit: (systemBehaviorGroup: Partial<BehaviorGroup>) => void;
+}
+
+export const CreateEditBehaviorGroupModal: React.FunctionComponent<CreateEditModalProps> = (props) => {
+
+    const [ systemBehaviorGroup, setSystemBehaviorGroup ] = React.useState<Partial<BehaviorGroup>>(props.initialSystemBehaviorGroup ?? {});
+
+    const actionOption = [
+        <FormSelectOption key={ 0 } isPlaceholder label='Choose an action' />,
+        <FormSelectOption key={ systemBehaviorGroup?.id } label='Send an email to Users: All' value={ systemBehaviorGroup?.id } />,
+        <FormSelectOption key={ systemBehaviorGroup?.id } label='Send an email to Users: Admins' value={ systemBehaviorGroup?.id } />
+    ];
+
+    const handleChange = (value: string, event: React.FormEvent<HTMLInputElement> | React.FormEvent<HTMLTextAreaElement>
+        | React.FormEvent<HTMLSelectElement>) => {
+        const target = event.target as HTMLInputElement | HTMLSelectElement;
+        setSystemBehaviorGroup(prev => ({ ...prev, [target.name]: target.value }));
+    };
+
+    const onSubmitLocal = React.useCallback(() => {
+        props.onSubmit(systemBehaviorGroup);
+    }, [ systemBehaviorGroup, props ]);
+
+    React.useEffect(() => {
+        setSystemBehaviorGroup(props.initialSystemBehaviorGroup ?? {});
+    }, [ props.initialSystemBehaviorGroup ]);
+
+    return (
+        <React.Fragment>
+            <Modal
+                variant={ ModalVariant.medium }
+                title={ `${ props.isEdit ? 'Update' : 'Create'} your System Behavior Group` }
+                isOpen={ props.showModal }
+                onClose={ props.onClose }
+            ><Form isHorizontal>
+                    <FormGroup label='Group Name' fieldId='displayName' isRequired
+                        helperText={ props.isEdit ? <HelperText><HelperTextItem variant="warning" hasIcon>
+                                                    If this field is modified it may affect exisiting behavior.
+                        </HelperTextItem></HelperText> : 'Enter a name for your group' }>
+                        <TextInput
+                            type='text'
+                            value={ systemBehaviorGroup.displayName }
+                            onChange={ handleChange }
+                            id='displayName'
+                            name="displayName"
+                        /></FormGroup>
+                    <FormGroup label='Action' fieldId='actions' isRequired>
+                        <FormSelect
```

Comment:
Still not allowing me to actually select an item and it show up in the table.

Reply:
Could you elaborate? What is actually doing and what do you want here?

Reply:
I want to be able to select an action from the drop down list of actions and it fill the field with the selected action. The dropdown shows list of actions but upon clicking on the action, it does not fill the field with selected action.

Reply:
You can see it in the video below.

Reply:
Is it doing the API call to update the server?